File: app/src/main/java/app/wt/noolis/model/Note.java
package app.wt.noolis.model;

import java.io.Serializable;

public class Note implements Serializable {
    private long id;
    private String tittle;
    private String content;
    private long last_edit;
    private int favourite = 0;
    private Category category;

    public Note() {
    }

    public Note(String tittle, String content, long last_edit, int favourite, Category category) {
        this.tittle = tittle;
        this.content = content;
        this.last_edit = last_edit;
        this.favourite = favourite;
        this.category = category;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTittle() {
        return tittle;
    }

    public void setTittle(String tittle) {
        this.tittle = tittle;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public long getLastEdit() {
        return last_edit;
    }

    public void setLastEdit(long last_edit) {
        this.last_edit = last_edit;
    }

    public int getFavourite() {
        return favourite;
    }

    public void setFavourite(int favourite) {
        this.favourite = favourite;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public void clear() {
        this.id = 0;
        this.tittle = null;
        this.content = null;
        this.last_edit = 0;
        this.favourite = 0;
    }
}

File: app/src/main/java/app/wt/noolis/model/Note.kt
package app.wt.noolis.model

import java.io.Serializable

class Note : Serializable {
    var id: Long = 0
    var tittle: String? = null
    var content: String? = null
    var lastEdit: Long = 0
    var favourite = 0
    var category: Category? = null

    constructor()

    constructor(tittle: String, content: String, last_edit: Long, favourite: Int, category: Category) {
        this.tittle = tittle
        this.content = content
        this.lastEdit = last_edit
        this.favourite = favourite
        this.category = category
    }

    fun clear() {
        this.id = 0
        this.tittle = null
        this.content = null
        this.lastEdit = 0
        this.favourite = 0
    }
}
